### PR TITLE
Add NuRaft logs

### PR DESCRIPTION
**Pull request by as51340**
### Description

- Added logger object for NuRaft logs which serves as adapter between spdlog and NuRaft logger :heavy_check_mark: 
- Logger object is shared between state machine, state manager and raft server :heavy_check_mark: 
- Existing logs from state machine and state manager are moved under this new logger :heavy_check_mark: 
- `CoordinatorInstanceInitConfig` extended with `nuraft_log_file` :heavy_check_mark: 
   - if arg not provided, logger will still be created but without any sinks, effectively creating no-op logger :heavy_check_mark: 
- Added `nuraft_log_file` flag :heavy_check_mark: 

### Logs examples
```
[2024-05-31 13:43:06.386] [NuRaft] [info] [asio_service.cxx:771] Raft ASIO listener initiated, UNSECURED
[2024-05-31 13:43:06.386] [NuRaft] [trace] [coordinator_state_machine.cpp:216] Getting last committed index from state machine.
[2024-05-31 13:43:06.386] [NuRaft] [trace] [coordinator_state_machine.cpp:216] Getting last committed index from state machine.
[2024-05-31 13:43:06.386] [NuRaft] [trace] [coordinator_state_machine.cpp:216] Getting last committed index from state machine.
[2024-05-31 13:43:06.386] [NuRaft] [trace] [coordinator_state_manager.cpp:100] Reading server state in coordinator state manager.
[2024-05-31 13:43:06.386] [NuRaft] [trace] [coordinator_state_manager.cpp:49] Loading cluster config from RocksDb
[2024-05-31 13:43:06.386] [NuRaft] [trace] [coordinator_state_manager.cpp:52] Didn't find anything stored on disk for cluster config.
[2024-05-31 13:43:06.386] [NuRaft] [trace] [coordinator_state_machine.cpp:204] Getting last snapshot from state machine.
[2024-05-31 13:43:06.386] [NuRaft] [trace] [coordinator_state_machine.cpp:207] There is no snapshot.
[2024-05-31 13:43:06.386] [NuRaft] [info] [raft_server.cxx:427] parameters: timeout 200 - 400, heartbeat 100, leadership expiry 200, max batch 100, backoff 50, snapshot distance 5, enable randomized snapshot creation NO, log sync stop gap 99999, reserved logs 5, client timeout 3000, auto forwarding OFF, API call type BLOCKING, custom commit quorum size 0, custom election quorum size 0, snapshot receiver INCLUDED, leadership transfer wait time 0, grace period of lagging state machine 0, snapshot IO: BLOCKING, parallel log appending: OFF
[2024-05-31 13:43:06.386] [NuRaft] [info] [raft_server.cxx:352] new timeout range: 200 -- 400
[2024-05-31 13:43:06.387] [NuRaft] [info] [raft_server.cxx:172]    === INIT RAFT SERVER ===
commit index 0
term 0
election timer allowed
log store start 1, end 0
config log idx 0, prev log idx 0
[2024-05-31 13:43:06.387] [NuRaft] [info] [raft_server.cxx:265] peer 1: DC ID 0, 0.0.0.0:10111, voting member, 1
my id: 1, voting_member
num peers: 0
[2024-05-31 13:43:06.387] [NuRaft] [info] [raft_server.cxx:284] global manager does not exist. will use local thread for commit and append
[2024-05-31 13:43:06.387] [NuRaft] [info] [raft_server.cxx:311] wait for HB, for 50 + [200, 400] ms
[2024-05-31 13:43:06.387] [NuRaft] [trace] [handle_commit.cxx:123] commit_cv_ sleep
[2024-05-31 13:43:06.387] [NuRaft] [info] [handle_append_entries.cxx:49] bg append_entries thread initiated
[2024-05-31 13:43:06.437] [NuRaft] [trace] [handle_timeout.cxx:184] re-schedule election timer
[2024-05-31 13:43:06.437] [NuRaft] [debug] [raft_server.cxx:319] server 1 started
[2024-05-31 13:43:06.437] [NuRaft] [trace] [asio_service.cxx:241] asio rpc session created: 0x728201401c10
[2024-05-31 13:43:06.800] [NuRaft] [trace] [handle_timeout.cxx:201] election timeout
[2024-05-31 13:43:06.800] [NuRaft] [warning] [handle_timeout.cxx:286] Election timeout, initiate leader election
[2024-05-31 13:43:06.800] [NuRaft] [info] [handle_priority.cxx:214] [PRIORITY] decay, target 1 -> 1, mine 1
[2024-05-31 13:43:06.800] [NuRaft] [info] [handle_timeout.cxx:304] [ELECTION TIMEOUT] current role: follower, log last term 0, state term 0, target p 1, my p 1, hb dead, pre-vote NOT done
[2024-05-31 13:43:06.800] [NuRaft] [trace] [coordinator_state_manager.cpp:94] Saving server state in coordinator state manager.
[2024-05-31 13:43:06.800] [NuRaft] [info] [handle_vote.cxx:230] [VOTE INIT] my id 1, my role candidate, term 1, log idx 0, log term 0, priority (target 1 / mine 1)
[2024-05-31 13:43:06.800] [NuRaft] [info] [raft_server.cxx:984] number of pending commit elements: 0
[2024-05-31 13:43:06.800] [NuRaft] [info] [raft_server.cxx:1000] state machine commit index 0, precommit index 0, last log index 0
[2024-05-31 13:43:06.800] [NuRaft] [info] [raft_server.cxx:1039] [BECOME LEADER] appended new config at 1
[2024-05-31 13:43:06.800] [NuRaft] [debug] [handle_commit.cxx:45] trigger commit upto 1
[2024-05-31 13:43:06.801] [NuRaft] [trace] [handle_commit.cxx:63] local log idx 1, target_commit_idx 1, quick_commit_index_ 1, state_->get_commit_idx() 0
[2024-05-31 13:43:06.801] [NuRaft] [trace] [handle_commit.cxx:74] commit_cv_ notify (local thread)
[2024-05-31 13:43:06.801] [NuRaft] [trace] [handle_commit.cxx:126] commit_cv_ wake up
[
```
- [x] Provide the full content or a guide for the final git message
    - **Add NuRaft logs**

### CI Testing Labels
Please select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [x] Write a release note, including added/changed clauses
    - **Added flag `--nuraft-log-file` for explicitly managing logs coming from NuRaft.**
- [x] Link the documentation PR here
    - **https://github.com/memgraph/documentation/pull/816**
- [x] Tag someone from docs team in the comments

### CODE REVIEW

File: src/coordination/include/nuraft/logger.hpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+ private:
+  /**
+   NuRaft log level:
+   *    Trace:    6
+   *    Debug:    5
+   *    Info:     4
+   *    Warning:  3
+   *    Error:    2
+   *    Fatal:    1
```
Comment (Ignition) [inline src/coordination/include/nuraft/logger.hpp:71]: use an enum

**Review by Ignition — CHANGES_REQUESTED** [review]

**Review by Ignition — APPROVED** [review]

**Comment by as51340** [comment]
@kgolubic RN: Added flag `--nuraft-log-file` for explicitly managing logs coming from NuRaft.

**Review by antoniofilipovic — APPROVED** [review]
Looks good to me

**Comment by kgolubic** [comment]
This will be RN:

- A new flag, `--nuraft-log-file`, has been added to explicitly manage logs
  coming from NuRaft.
[#2084](https://github.com/memgraph/memgraph/pull/2084)

**Pull request merged**